tensorflow_probability/python/mcmc/transformed_kernel_test.py

```diff
@@ -308,6 +326,89 @@ def test_is_calibrated(self):
             FakeInnerKernel(lambda x: -x**2 / 2, False),
             tfb.Identity()).is_calibrated)
 
+  def test_bijector_valid_transform_then_step(self):
+    new_kernel = transform_then_step(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.),
+                    pkr_one.inner_results.transformed_state,
+                    err=1e-6)
+    self.assertNear(np.log(9.),
+                    pkr_two.inner_results.transformed_state,
+                    err=1e-6)
+
+  def test_bijector_valid_step_then_transform(self):
+    new_kernel = step_then_transform(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.), pkr_one.transformed_state, err=1e-6)
+    self.assertNear(np.log(9.), pkr_two.transformed_state, err=1e-6)
+
+  def test_step_size_changed(self):
+    target_dist = tfd.MultivariateNormalDiag(loc=[0., 0.], scale_diag=[1., 10.])
+    # `hmc_kernel`'s step size is far from optimal
+    hmc_kernel = tfp.mcmc.HamiltonianMonteCarlo(
+        target_log_prob_fn=target_dist.log_prob,
+        num_leapfrog_steps=27,
+        step_size=10,
+        seed=_maybe_seed(test_util.test_seed()))
+    step_adaptation_kernel = tfp.mcmc.SimpleStepSizeAdaptation(
+        inner_kernel=hmc_kernel,
+        adaptation_rate=0.8,
+        num_adaptation_steps=9)
+    trans_kernel = tfp.mcmc.TransformedTransitionKernel(
+        inner_kernel=step_adaptation_kernel,
+        bijector=tfb.Exp()
+    )
+    kernel_results = trans_kernel.inner_kernel.bootstrap_results(tf.zeros(2))
+    for _ in range(2):
+      _, kernel_results = trans_kernel.inner_kernel.one_step(tf.zeros(2),
+                                                             kernel_results)
+    adapted_step_size = self.evaluate(
+        kernel_results.inner_results.accepted_results.step_size)
+    self.assertLess(adapted_step_size, 7)
+
+  def test_deeply_nested(self):
```

Comment:
Um, shouldn't this test put the transformed transition kernel on the outside, to check that it penetrates multiple levels of inner kernels?

Reply:
Yeah, good point.